internal/builder/builder_test.go

```diff
@@ -1846,6 +1846,32 @@ func testBuilder(t *testing.T, when spec.G, it spec.S) {
 				})
 			})
 		})
+
+		when("#New", func() {
+			when("#WithRunImage", func() {
+				// Current runImage information in builder image:
+				// "stack": {"runImage": {"image": "prev/run", "mirrors": ["prev/mirror"]}}
+				var newBuilder *builder.Builder
+				newRunImage := "another/run"
+
+				it.Before(func() {
+					var err error
+					newBuilder, err = builder.New(builderImage, "newBuilder/image", builder.WithRunImage(newRunImage))
+					h.AssertNil(t, err)
+				})
+
+				it("Overrides de RunImageMetadata", func() {
```

Comment:
Maybe we should clarify somewhere (in a code comment or something) that pack always recreates stack.toml and run.toml from the metadata when creating the ephemeral builder. Because we're not just overriding the metadata, we're changing what the lifecycle encounters on the filesystem.